File: src/contests/cg/puzzle/python3/max-area.py
# 🎮 CodinGame Puzzle - max-area
# https://www.codingame.com/training/easy/max-area

input()
a = list(map(int, input().split()))
left = 0
right = len(a) - 1
best = 0
while left < right:
    height = min(a[left], a[right])
    area = height * (right - left)
    if area > best:
        best = area
    if a[left] < a[right]:
        left += 1
    else:
        right -= 1
print(best)

File: src/contests/cg/puzzle/ruby/max-area.rb
# 🎮 CodinGame Puzzle - max-area
# https://www.codingame.com/training/easy/max-area

gets
a = gets.split.map(&:to_i)
left = 0
right = a.length - 1
best = 0
while left < right
  height = [a[left], a[right]].min
  area = height * (right - left)
  best = area if area > best
  if a[left] < a[right]
    left += 1
  else
    right -= 1
  end
end
puts best
